### ISSUE TITLE
[Feature Request] Parser for generic key=value pairs

### ISSUE THREAD

**Issue by 1848** [influxdata/telegraf#5446]
Hi, I need a parser which reads multiple lines of key=value pairs and translates them into metrics.  
I am using the http input plugin to read something like
```plain
key1=2.456
keyB=455
keyX=whatever
```
I want every key to be converted into a metric but I could not find any parser which provides this capability, is there any?

I made this (https://github.com/1848/telegraf/commit/80fd23be4280f568cc6eb83dad2045675070e74d) generic parser which gets the job done but I dont know golang good enough to sanitize this.

Is there already a plugin (which I did not notice) which solves my problem?
If not, is there anyone who can (and wants to) implement this?

Thanks for reading. 

PS:
I noticed there is #4466 which is similar but uses pattern matching which I dont need.

**Comment by danielnelson** [comment]
You should be able to use the [logfmt parser](https://github.com/influxdata/telegraf/tree/master/plugins/parsers/logfmt) for this.

**Comment by gsauthof** [comment]
@danielnelson the issue with the logfmt parser is that it doesn't support parsing multiple key/value lines into one metric with all the key/value pairs as fields and/or tags.

In contrast, the [grok parser](https://github.com/influxdata/telegraf/tree/master/plugins/parsers/grok) supports such multiline parsing.
However, in case when some keys are optional and the order of the keys isn't fixed such a grok expression quickly grows very ugly and is hard to maintain.

Since Linux provides some metrics in such `key=value`, `key value` or `key: value` or similar key/value per line formats, better parsing support is needed, arguably.

For example the kernel provides some metrics under `/proc` and `/sys`,
such as `/sys/devices/system/node/node0/numastat`, `/sys/devices/system/node/node0/vmstat` , `/proc/vmstat` or even some metrics provided by daemons.

Basically, these files are transposed csv files, so perhaps it would even make sense to extend the csv parser such that field names can be read from the first and their values from the following column.
